src/materials/SteelCreepDamage.C

```diff
@@ -0,0 +1,189 @@
+  params.addRangeCheckedParam<Real>("initial_damage_value",
+                                    0.9,
+                                    "initial_damage_value <= 1.0 & initial_damage_value >= 0.0",
+                                    "Initial value of damage that will trigger linear reduction of "
+                                    "quadrature point's load-carrying capacity.");
+  params.addParam<std::string>("creep_strain_name",
+                               "creep_strain",
+                               "Name of the creep strain material driving damage failure.");
+  return params;
+}
+
+SteelCreepDamage::SteelCreepDamage(const InputParameters & parameters)
+  : ScalarDamageBase(parameters),
+    GuaranteeConsumer(this),
+    _creep_strain_name(getParam<std::string>("creep_strain_name")),
+    _creep_strain(getMaterialProperty<RankTwoTensor>(_base_name + _creep_strain_name)),
+    _creep_strain_old(getMaterialPropertyOld<RankTwoTensor>(_base_name + _creep_strain_name)),
+    _epsilon_f(getParam<Real>("epsilon_f")),
+    _reduce_factor(getParam<Real>("reduce_factor")),
+    _initial_damage_value(getParam<Real>("initial_damage_value")),
+    _creep_law_exponent(getParam<Real>("creep_law_exponent")),
+    _stress(getMaterialProperty<RankTwoTensor>(_base_name + "stress"))
+{
+}
+
+void
+SteelCreepDamage::initQpStatefulProperties()
+{
+  _damage_index[_qp] = 0.0;
+}
+
+const Real &
+SteelCreepDamage::getQpDamageIndex(unsigned int /*qp*/)
+{
+  return _damage_index[_qp];
+}
+
+void
+SteelCreepDamage::updateDamage()
+{
+  updateQpDamageIndex();
+}
+
+void
+SteelCreepDamage::updateQpDamageIndex()
+{
+  Real epsilon_f_star;
+  const Real vonMises = RankTwoScalarTools::vonMisesStress(_stress[_qp]);
+
+  if (vonMises > 1.0e-6)
+  {
+    const Real h = RankTwoScalarTools::triaxialityStress(_stress[_qp]);
+
+    // Update creep ductility due to triaxiliaty effects
+    epsilon_f_star =
+        _epsilon_f *
+        std::sinh(2.0 / 3.0 * (_creep_law_exponent - 0.5) / (_creep_law_exponent + 0.5)) /
+        std::sinh(2.0 * (_creep_law_exponent - 0.5) / (_creep_law_exponent + 0.5) * std::abs(h));
+  }
+  else
+    epsilon_f_star = _epsilon_f;
+
+  RankTwoTensor creep_increment = _creep_strain[_qp] - _creep_strain_old[_qp];
+
+  // Get equivalent creep strain increment
+  Real equivalent_creep_increment =
+      1 / std::sqrt(2) *
+      std::sqrt(Utility::pow<2>(creep_increment(0, 0) - creep_increment(1, 1)) +
+                Utility::pow<2>(creep_increment(1, 1) - creep_increment(2, 2)) +
+                Utility::pow<2>(creep_increment(2, 2) - creep_increment(0, 0)) +
+                1.5 * creep_increment(0, 1) * creep_increment(0, 1) +
+                1.5 * creep_increment(1, 2) * creep_increment(1, 2) +
+                1.5 * creep_increment(2, 0) * creep_increment(2, 0));
+
+  _damage_index[_qp] = _damage_index_old[_qp] + _dt * equivalent_creep_increment / epsilon_f_star;
```

Comment:
do an assertion for `epsilon_f_star != 0`

Reply:
Yeah, will do. I am trying to figure out why I am getting it...